[build.webkit.org] Upload jscore-test logs to S3

briannafan opened the pull request:
#### 1bcae33cfb73db9cdadcca1a30f87720342464ec
<pre>
[build.webkit.org] Upload jscore-test logs to S3
<a href="https://bugs.webkit.org/show_bug.cgi?id=270677">https://bugs.webkit.org/show_bug.cgi?id=270677</a>
<a href="https://rdar.apple.com/problem/124654490">rdar://problem/124654490</a>

Reviewed by Jonathan Bedard.

Filter JSC test logs and upload full logs to S3.

* Tools/CISupport/build-webkit-org/steps.py:
(RunJavaScriptCoreTests.run):
(RunJavaScriptCoreTests):
(RunJavaScriptCoreTests.commandComplete):

Canonical link: <a href="https://commits.webkit.org/276960@main">https://commits.webkit.org/276960@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5469df8f02578a813fa15c5e8b2d5e9db26d0bea

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46086 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25216 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48671 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48757 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42126 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/48393 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29676 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22658 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37686 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46664 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22411 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39763 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18876 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19800 "Found unexpected failure with change (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/40861 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4130 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/39319 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42508 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41211 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50549 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/45560 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/21082 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17602 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44851 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/46125 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22382 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43764 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22741 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/52711 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6452 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22076 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10782 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/Scripts/filter-jsc-tests.py:
```diff
@@ -0,0 +1,56 @@
+    log_file = os.path.abspath(f'{log_file_name}')
+    with open(log_file, "w") as f:
+        for line in sys.stdin:
+            if not line.startswith('Running') and not line.startswith('Skipping'):
```
Please check how much is the number of output lines after that filtering. For example: for this run with 500k+ lines, how much is the number of lines of output https://build.webkit.org/#/builders/64/builds/225239

briannafan replied:
Oh that's a good point. I changed the filtering so it only prints lines after a line starts with `**` or `Results`, which indicates all tests have finished and results are being printed.

aj062 commented on Tools/Scripts/filter-jsc-tests.py:
```diff
@@ -0,0 +1,56 @@
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
```
Since we are adding a script which will run on bots, it's better to land it first and then deploy the change to buildbot after few days. so that when your change is picked up by buildbot, the script is already in repository. It's possible that buildbot is testing an old revision for which this script won't exist. So better to have few days interval between adding this script and restarting buildbot to deploy a change which uses this script.

briannafan replied:
Created a separate PR so we can land this first: https://github.com/WebKit/WebKit/pull/26683

aj062 commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -718,6 +718,14 @@ def __init__(self, *args, **kwargs):
             kwargs['sigtermTime'] = 10
         super().__init__(*args, **kwargs)
 
+    def buildCommandKwargs(self, warnings):
```
Do we need this method? Can we do this in __init__?

briannafan replied:
Unfortunately build properties cannot be accessed in `init`

aj062 commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -742,6 +750,21 @@ def run(self):
             self.command += ['--test-writer=ruby']
 
         self.appendCustomBuildFlags(platform, self.getProperty('fullPlatform'))
+        self.command = ['/bin/sh', '-c', ' '.join(self.command) + ' 2>&1 | python3 Tools/Scripts/filter-jsc-tests.py']
```
Does it work fine with WithProperties?

briannafan replied:
Yeah, `self.command` is handled by buildbot

aj062 commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -762,6 +785,21 @@ def countFailures(self):
 
         return count
 
+    def commandComplete(self, cmd):
```
Do we need this method? (Especially that we are using run () method)

briannafan replied:
Oh I already moved the code to `run()` as seen above, I forgot to delete this oops.

JonWBedard reviewed (approved):
This should land after #26683

briannafan committed:
[build.webkit.org] Upload jscore-test logs to S3
https://bugs.webkit.org/show_bug.cgi?id=270677
rdar://problem/124654490

Reviewed by Jonathan Bedard.

Filter JSC test logs and upload full logs to S3.

* Tools/CISupport/build-webkit-org/steps.py:
(RunJavaScriptCoreTests.run):
(RunJavaScriptCoreTests):
(RunJavaScriptCoreTests.commandComplete):

Canonical link: https://commits.webkit.org/276960@main
